contracts/peg/VanillaPriceBoundPSM.sol

```diff
@@ -0,0 +1,131 @@
+/// setOracleFloor and setOracleCeiling
+contract VanillaPriceBoundPSM is VanillaPSM, IPriceBound {
+    using Decimal for Decimal.D256;
+    using SafeERC20 for IERC20;
+    using SafeCast for *;
+
+    /// @notice the default minimum acceptable oracle price floor is 98 cents
+    uint256 public override floor;
+
+    /// @notice the default maximum acceptable oracle price ceiling is $1.02
+    uint256 public override ceiling;
+
+    /// @notice constructor
+    /// @param _floor minimum acceptable oracle price
+    /// @param _ceiling maximum  acceptable oracle price
+    /// @param _params PSM construction params
+    constructor(
+        uint256 _floor,
+        uint256 _ceiling,
+        OracleParams memory _params,
+        uint256 _reservesThreshold,
+        IERC20 _underlyingToken,
+        IPCVDeposit _surplusTarget
+    )
+        VanillaPSM(
+            _params,
+            _reservesThreshold,
+            _underlyingToken,
+            _surplusTarget
+        )
+    {
+        _setCeilingBasisPoints(_ceiling);
+        _setFloorBasisPoints(_floor);
+    }
+
+    /// @notice sets the floor price in BP
+    function setOracleFloorBasisPoints(uint256 newFloorBasisPoints)
+        external
+        override
+        onlyGovernorOrAdmin
+    {
+        _setFloorBasisPoints(newFloorBasisPoints);
+    }
+
+    /// @notice sets the ceiling price in BP
+    function setOracleCeilingBasisPoints(uint256 newCeilingBasisPoints)
+        external
+        override
+        onlyGovernorOrAdmin
+    {
+        _setCeilingBasisPoints(newCeilingBasisPoints);
+    }
+
+    function isPriceValid() external view override returns (bool) {
+        return _validPrice(readOracle());
+    }
+
+    /// @notice helper function to set the ceiling in basis points
+    function _setCeilingBasisPoints(uint256 newCeilingBasisPoints) internal {
+        require(
+            newCeilingBasisPoints != 0,
+            "PegStabilityModule: invalid ceiling"
+        );
+        require(
+            Decimal
+                .ratio(
+                    newCeilingBasisPoints,
+                    Constants.BASIS_POINTS_GRANULARITY
+                )
+                .greaterThan(
+                    Decimal.ratio(floor, Constants.BASIS_POINTS_GRANULARITY)
+                ),
+            "PegStabilityModule: ceiling must be greater than floor"
+        );
+        uint256 oldCeiling = ceiling;
+        ceiling = newCeilingBasisPoints;
+
+        emit OracleCeilingUpdate(oldCeiling, ceiling);
+    }
+
+    /// @notice helper function to set the floor in basis points
+    function _setFloorBasisPoints(uint256 newFloorBasisPoints) internal {
+        require(newFloorBasisPoints != 0, "PegStabilityModule: invalid floor");
+        require(
+            Decimal
+                .ratio(newFloorBasisPoints, Constants.BASIS_POINTS_GRANULARITY)
+                .lessThan(
+                    Decimal.ratio(ceiling, Constants.BASIS_POINTS_GRANULARITY)
+                ),
```

Comment:
let's see if we can just use a regular `<` operator here instead of using decimal ratio